lib/ssl/src/dtls_socket.erl

```diff
@@ -69,12 +69,13 @@ listen(Port, #config{inet_ssl = SockOpts,
             Error
     end.
 
-accept({Listener,_}, #config{transport_info = Info,
-                             connection_cb = ConnectionCb}, _Timeout) ->
-    Transport = element(1, Info),
+accept({Listener,_}, #config{}, _Timeout) ->
     case dtls_packet_demux:accept(Listener, self()) of
-	{ok, Pid, Socket} ->
-	    {ok, socket([Pid], Transport, {Listener, Socket}, ConnectionCb)};
+	{ok, Pid, _Socket} ->
```

Comment:
This function is only called here. I do not think we need the third argument.  

The old socket function looked like:

```
socket(Pids, gen_udp = Transport,
       PeerAndSock = {{_Host, _Port}, _Socket}, ConnectionCb) ->
    #sslsocket{pid = Pids, 
	       %% "The name "fd" is kept for backwards compatibility
	       fd = {Transport, PeerAndSock, ConnectionCb}};
socket(Pids, Transport, Socket, ConnectionCb) ->
    #sslsocket{pid = Pids, 
	       %% "The name "fd" is kept for backwards compatibility
	       fd = {Transport, Socket, ConnectionCb}}.

```

But I in practice this would accomplish nothing other than maybe documentation hint that udp sockets do not  have connections so we need more info to emulate the connection.

Reply:
Fixed